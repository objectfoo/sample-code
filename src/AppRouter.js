import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import {pageMap} from './pages/config';

export const App = ({classes}) => {
  const orderedPages = getOrderedPages(pageMap);

  return (
    <Router>
      <div>
        {orderedPages.map(({key, path, exact, ...rest}) => {
          const routeP = { key, path, exact };
          return (
            <Route {...routeP} render={() => (
              <rest.component pages={orderedPages} currentKey={key} />
            )} />
          );
        })}
      </div>
    </Router>
  );
};

export default App;

export const getOrderedPages = pages =>
  pages.ordered.map(key => pages.byKey[key]);
