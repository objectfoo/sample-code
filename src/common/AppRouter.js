import React from 'react';
import {BrowserRouter as Router, Route} from 'react-router-dom';
import {pageMap} from './config';

export const AppRouter = () => {
  const orderedPages = getOrderedPages(pageMap);

  return (
    <Router>
      <div>
        {orderedPages.map(({key, path, exact, ...rest}) => {
          const routeP = { key, path, exact };
          return (
            <Route {...routeP} render={() => (
              <rest.component pages={orderedPages} currentKey={key} />
            )} />
          );
        })}
      </div>
    </Router>
  );
};

export default AppRouter;

export const getOrderedPages = pages =>
  pages.ordered.map(key => pages.byKey[key]);
