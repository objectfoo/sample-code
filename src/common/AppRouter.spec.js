import React from 'react';
import {shallow} from 'enzyme';
import {AppRouter} from './AppRouter';

describe('AppRouter', () => {
  it('should route', () => {
    expect(shallow(<AppRouter />)).toMatchSnapshot();
  });
});
