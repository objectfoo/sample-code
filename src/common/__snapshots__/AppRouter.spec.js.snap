// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter should route 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppRouter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <Route
                exact={true}
                path="/"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/code-sample-example"
                render={[Function]}
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/code-sample-example"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "home",
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "code-sample-example",
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/code-sample-example",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/code-sample-example"
                    render={[Function]}
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/code-sample-example"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "home",
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "code-sample-example",
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/code-sample-example",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
